contextualized/regression/tests.py

```diff
@@ -56,29 +56,33 @@ def setUp(self):
         self.c_dim, self.x_dim, self.y_dim = c_dim, x_dim, y_dim
         self.C, self.X, self.Y = C.numpy(), X.numpy(), Y.numpy()
 
-    def _quicktest(self, model, univariate=False, correlation=False):
+    def _quicktest(self, model, univariate=False, correlation=False, markov=False):
         print(f'\n{type(model)} quicktest')
-        dataloader = None
-        trainer = None
         if correlation:
             dataloader = model.dataloader(self.C, self.X, batch_size=self.batch_size)
             trainer = CorrelationTrainer(max_epochs=self.epochs)
+            y_true = np.tile(self.X[:,:,np.newaxis], (1, 1, self.X.shape[-1]))
+        elif markov:
+            dataloader = model.dataloader(self.C, self.X, batch_size=self.batch_size)
+            trainer = MarkovTrainer(max_epochs=self.epochs)
+            y_true = self.X
         else:
             dataloader = model.dataloader(self.C, self.X, self.Y, batch_size=self.batch_size)
             trainer = RegressionTrainer(max_epochs=self.epochs)
+            if univariate:
+                y_true = np.tile(self.Y[:, :, np.newaxis], (1, 1, self.X.shape[-1]))
+            else:
+                y_true = self.Y
         y_preds = trainer.predict_y(model, dataloader)
-        y_true = self.Y
-        if univariate:
-            y_true = np.tile(y_true[:,:,np.newaxis], (1, 1, self.X.shape[-1]))
-        if correlation:
-            y_true = np.tile(self.X[:,:,np.newaxis], (1, 1, self.X.shape[-1]))
         err_init = ((y_true - y_preds)**2).mean()
         trainer.fit(model, dataloader)
         trainer.validate(model, dataloader)
         trainer.test(model, dataloader)
         beta_preds, mu_preds = trainer.predict_params(model, dataloader)
         if correlation:
             rhos = trainer.predict_correlation(model, dataloader)
+        if markov:
+            omegas = trainer.predict_precision(model, dataloader)
```

Comment:
Are the values returned here tested? Or is just testing that the `predict_` functions return without error?

Reply:
Just to make sure there are no errors. It would be nice to have some test cases that check if we recover true models, but currently tests just check convergence in MSE and if any components throw errors.